src/nvim/ex_getln.c

```diff
@@ -4444,6 +4448,15 @@ static int open_cmdwin(void)
   RedrawingDisabled = 0;
   int save_count = save_batch_count();
 
+  MAXSIZE_TEMP_DICT(opts, 1);
+  PUT_C(opts, "line_based", BOOLEAN_OBJ(true));
+  MAXSIZE_TEMP_ARRAY(args, 3);
+  ADD_C(args, INTEGER_OBJ(cmdwin_buf->handle));
+  ADD_C(args, CSTR_TO_OBJ("vim"));
+  ADD_C(args, DICTIONARY_OBJ(opts));
+  Error err = ERROR_INIT;
+  NLUA_EXEC_STATIC("vim.treesitter.start(...)", args, kRetNilBool, NULL, &err);
```

Comment:
This probably should only apply to specific `histtypes`, like how the old logic set the cmdwin's `&filetype` before?

For example, you probably don't want a cmdwin opened for `input()` via CTRL-F to highlight everything as Vim script, as the input could be for any purpose.